src/shelter/use-shelter.hook.ts

```diff
@@ -56,15 +57,32 @@ export const useShelter = () => {
           }
         }),
       createImportedAccount$
-        .pipe(switchMap(({ privateKey, name }) => Shelter.createImportedAccount$(privateKey, name)))
         .pipe(
-          catchError(() => {
-            showErrorToast({
-              title: 'Failed to import account.',
-              description: 'This may happen because provided Key is invalid.'
-            });
+          switchMap(({ privateKey, name }) => {
+            return getPublicKeyAndHash$(privateKey)
+              .pipe(
+                switchMap(([publicKey]) => {
+                  for (const account of accounts) {
+                    if (account.publicKey === publicKey) {
+                      showWarningToast({ description: 'Account already exist' });
```

Comment:
move "account already exist logic" into a component where `createImportedAccount` is called

Reply:
i'm not agree. Here's one place with account checking, and we don't need to reuse shelter method to define `publicKey` and check if account exist. In your case - we should make some complicated steps on submit. Also,  `getPublicKeyAndHash` method is observable. What do you think about this?